src/components/pages/TodoPage.js

```diff
@@ -8,35 +8,40 @@ class TodoPage extends Component {
 
   render() {
 
+    const { todos } = this.props
+
     return (
       <div>
         <h1 className="placeholder-page-label"> Todo Page </h1>
         <table className="ui celled striped table todos-table">
           <thead>
             <tr>
-              <th>
-                Title
-              </th>
-              <th>
-                Description
-              </th>
-              <th>
-                Completed
-              </th>
+              <th> Title </th>
+              <th> Description </th>
+              <th> Completed </th>
             </tr>
           </thead>
           <tbody>
             <tr>
-              <td>
-                Get weird
-              </td>
-              <td>
-                Let's get weird
-              </td>
-              <td>
-
-              </td>
+              <td> Get weird </td>
+              <td> Lets get weird </td>
+              <td> </td>
             </tr>
+
+            {
+              Object.keys(todos).map(id => {
```

Comment:
There are a couple things going on here.  First within the JSX part of a component, you can add logic inside of these ```{ }```

Within these I have access to all the component data, including the todos I just got out of this.props

Todos is an object.  So I want to get all of the keys with Object.keys

then I want to use those keys to get the actual objects: todos[id]

then I return a <tr> filled with data from the todo I just accessed.

You should look up the Array.map function if you are unfamiliar, as well as Object.keys

Reply:
how come the first object of the table is still there then? Can we not leave the logic to fill out the tables and call that method in from another page? Actually how do we use this logic to make more objects in the todos?

Reply:
oooo do we use array.map function to add items to the todos array?